tutorials/Building a Custom Environment in Aviary.ipynb

```diff
@@ -0,0 +1,158 @@
+    "# State in this example is simply a counter\n",
+    "class CounterEnvState(BaseModel):\n",
+    "    count: int\n",
+    "\n",
+    "\n",
+    "class CounterEnv(Environment[CounterEnvState]):\n",
+    "    \"\"\"A simple environment that allows an agent to modify a counter.\"\"\"\n",
+    "\n",
+    "    async def reset(self):\n",
+    "        \"\"\"Initialize the environment with a counter set to 0.\"\"\"\n",
+    "        self.state = CounterEnvState(count=0)\n",
+    "\n",
+    "        # Create tools allowing the agent to increment and decrement counter\n",
+    "        self.tools = [\n",
+    "            Tool.from_function(self.incr),\n",
+    "            Tool.from_function(self.decr),\n",
+    "        ]\n",
+    "\n",
+    "        # Return an observation message with the counter and available tools\n",
+    "        return [Message(content=f\"counter={self.state.count}\")], self.tools\n",
+    "\n",
+    "    async def step(self, action: ToolRequestMessage):\n",
+    "        \"\"\"Executes the tool call requested by the agent.\"\"\"\n",
+    "        obs = await self.exec_tool_calls(action)\n",
+    "\n",
+    "        # The reward is the square of the current count\n",
+    "        reward = self.state.count**2\n",
+    "\n",
+    "        # Returns observations, reward, done, truncated\n",
+    "        return obs, reward, reward < 0, False\n",
+    "\n",
+    "    def incr(self):\n",
+    "        \"\"\"Increment the counter.\"\"\"\n",
+    "        self.state.count += 1\n",
+    "        return f\"counter={self.state.count}\"\n",
+    "\n",
+    "    def decr(self):\n",
+    "        \"\"\"Decrement the counter.\"\"\"\n",
+    "        self.state.count -= 1\n",
+    "        return f\"counter={self.state.count}\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "25a02b15-dd96-47e1-990f-7678f7469b4f",
+   "metadata": {},
+   "source": [
+    "## Evaluating an Agent on the Environment\n",
+    "\n",
+    "Following the definition of our custom environment, we can now evaluate a language agent\n",
+    "on the environment using Aviary's sister library LDP (https://github.com/Future-House/ldp)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "2332c261-0dc9-4137-a119-bd28132482b4",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import litellm\n",
+    "from ldp.agent import Agent\n",
+    "from ldp.alg.rollout import RolloutManager\n",
+    "from ldp.graph import LLMCallOp\n",
+    "\n",
+    "\n",
+    "class AgentState:\n",
+    "    \"\"\"A container for maintaining agent state across interactions.\"\"\"\n",
+    "\n",
+    "    def __init__(self, messages, tools):\n",
+    "        self.messages = messages\n",
+    "        self.tools = tools\n",
+    "\n",
+    "\n",
+    "class SimpleAgent(Agent):\n",
+    "    def __init__(self, **kwargs):\n",
+    "        self._llm_call_op = LLMCallOp(**kwargs)\n",
+    "\n",
+    "    async def init_state(self, tools):\n",
```

Comment:
Can you type hint these signatures for the demo code

Reply:
Done